packages/redux/src/raffles/__tests__/reducer.test.ts

```diff
@@ -0,0 +1,264 @@
+      actionTypes.UPDATE_RAFFLE_PARTICIPATION_FAILURE,
+    ])('should handle %s action type', actionType => {
+      const state = reducer(initialState, {
+        type: actionType,
+        payload: { error: expectedError },
+        meta: { raffleId, participationId },
+      });
+      expect(state.participations[participationId]?.isLoading).toBe(false);
+      expect(state.participations[participationId]?.error).toBe(expectedError);
+    });
+  });
+
+  describe('participation creation reducer', () => {
+    const expectedError = new Error('fetch participation creation error');
+
+    it(`should handle ${actionTypes.CREATE_RAFFLE_PARTICIPATION_REQUEST} action type`, () => {
+      const state = reducer(initialState, {
+        type: actionTypes.CREATE_RAFFLE_PARTICIPATION_REQUEST,
+        meta: { raffleId },
+      });
+      expect(state.participationCreations[raffleId]?.isLoading).toBe(true);
+      expect(state.participationCreations[raffleId]?.error).toBeNull();
+    });
+
+    it(`should handle ${actionTypes.CREATE_RAFFLE_PARTICIPATION_SUCCESS} action type`, () => {
+      const state = reducer(initialState, {
+        type: actionTypes.CREATE_RAFFLE_PARTICIPATION_SUCCESS,
+        payload: mockCreateRaffleParticipationsNormalizedPayload,
+        meta: { raffleId, participationId },
+      });
+      expect(state.participationCreations[raffleId]?.isLoading).toBe(false);
+      expect(state.participationCreations[raffleId]?.error).toBeNull();
+    });
+
+    it(`should handle ${actionTypes.CREATE_RAFFLE_PARTICIPATION_FAILURE} action type`, () => {
+      const state = reducer(initialState, {
+        type: actionTypes.CREATE_RAFFLE_PARTICIPATION_FAILURE,
+        payload: { error: expectedError },
+        meta: { raffleId },
+      });
+      expect(state.participationCreations[raffleId]?.isLoading).toBe(false);
+      expect(state.participationCreations[raffleId]?.error).toBe(expectedError);
+    });
+  });
+
+  describe('getAllRaffles selector', () => {
+    it('should return the state', () => {
+      expect(getAllRaffles(initialState)).toBe(initialState.allRaffles);
+    });
+  });
+
+  describe('getRaffles selector', () => {
+    it('should return the state', () => {
+      expect(getRaffles(initialState)).toBe(initialState.raffles);
+    });
+  });
+
+  describe('getEstimations selector', () => {
+    it('should return the state', () => {
+      expect(getEstimations(initialState)).toBe(initialState.estimations);
+    });
+  });
+
+  describe('getParticipations selector', () => {
+    it('should return the state', () => {
+      expect(getParticipations(initialState)).toBe(initialState.participations);
+    });
+  });
+
+  describe('getParticipationCreations selector', () => {
+    it('should return the state', () => {
+      expect(getParticipationCreations(initialState)).toBe(
+        initialState.participationCreations,
+      );
+    });
+  });
+
+  describe('entitiesMapper', () => {
+    const entityState = {
```

Comment:
Use a state that contains other entities. Imagine that the entitiesMapper for the `RESET_RAFFLES_STATE` returns `{}`. If that was the case, this test would not catch the bug.

Reply:
done, pls double check